Library/Homebrew/cmd/update-bash.sh

```diff
@@ -208,6 +208,12 @@ update-bash() {
     esac
   done
 
+  if ! curl -I --connect-timeout 2 https://github.com &> /dev/null
```

Comment:
Ok, leave it as `curl` :+1:. May be worth trying some other hosts like http://brew.sh or https://status.github.com for speed. I also think we need to make this timeout much higher (e.g. closer to what `git fetch` does by default).